### PR TITLE
feat(prefer-explicit-assert): report on `findBy*` queries too

### PR DESCRIPTION
**Pull request by zaicevas**
Part of https://github.com/testing-library/eslint-plugin-testing-library/issues/409

Reports standalone `findBy*` queries. Reporting wrong assertions is out of scope for this PR.

### CODE REVIEW

File: lib/rules/prefer-explicit-assert.ts [hunk 1]
```diff
@@ -55,14 +97,33 @@ export default createTestingLibraryRule<Options, MessageIds>({
...
           getQueryCalls.push(node);
         }
+
+        if (helpers.isFindQueryVariant(node)) {
+          findQueryCalls.push(node);
+        }
       },
       'Program:exit'() {
+        findQueryCalls.forEach((queryCall) => {
+          if (isVariableDeclaration(queryCall)) {
```
Comment (Belco90) [inline lib/rules/prefer-explicit-assert.ts:114]: Wouldn't be better to exclude those these nodes when pushing them to `findQueryCalls`?

File: tests/lib/rules/prefer-explicit-assert.test.ts [hunk 2]
```diff
@@ -54,6 +54,24 @@ ruleTester.run(RULE_NAME, rule, {
...
+    })),
+    ...COMBINED_QUERIES_METHODS.map((queryMethod) => ({
+      code: `const quxElement = screen.find${queryMethod}('qux')`,
+    })),
+    ...COMBINED_QUERIES_METHODS.map((queryMethod) => ({
+      code: `
+      async () => {
+        const quxElement = await screen.find${queryMethod}('qux')
+      }`,
+    })),
```
Comment (Belco90) [inline tests/lib/rules/prefer-explicit-assert.test.ts:74]: Can you add an extra valid test for a function returning a promise from `findBy*` query?

```js
function findBySubmit() {
  // this is valid
  return screen.findByRole('button', { name: 'submit' })
}
```

**Review by Belco90 — CHANGES_REQUESTED** [review]

File: tests/lib/rules/prefer-explicit-assert.test.ts (hunk 2 above)
Comment (zaicevas) [reply tests/lib/rules/prefer-explicit-assert.test.ts:74]: Oh, seems like you've found a bug 😄

Comment (zaicevas) [reply tests/lib/rules/prefer-explicit-assert.test.ts:74]: Great catch :) Should be good now

File: lib/rules/prefer-explicit-assert.ts (hunk 1 above)
Comment (zaicevas) [reply lib/rules/prefer-explicit-assert.ts]: Not sure about that 🤔 Now there's even more logic in this if statement, so technically we could move everything into `'CallExpression Identifier'(node: TSESTree.Identifier) {`, but I think the current version is more consistent with how `getBy` case is implemented.

**Review by Belco90 — APPROVED** [review]
Seems good to go 🚀 

Thanks for your work on the recent PRs!

File: lib/rules/prefer-explicit-assert.ts (hunk 1 above)
Comment (Belco90) [reply lib/rules/prefer-explicit-assert.ts]: Fair enough, we can leave it as it is.

**Pull request merged**